multinet/db.py

```diff
@@ -608,3 +608,44 @@ def count_text(filt: str) -> str:
         "edges": list(aql_query(workspace, query)),
         "count": next(aql_query(workspace, count)),
     }
+
+
+@lru_cache(maxsize=1)
+def uploads_collection() -> StandardCollection:
+    """Return the collection used for temporarily storing uploaded file chunks."""
+    sysdb = db("_system")
+
+    if not sysdb.has_collection("uploads"):
+        sysdb.create_collection("uploads")
+
+    return sysdb.collection("uploads")
+
+
+def create_upload_document() -> Any:
+    """Insert empty multipart upload temp document."""
+    # returns arango document key as upload_id for client
+    return uploads_collection().insert({}).get("_key")
+
+
+def insert_file_chunk(key: str, sequence: str, chunk: str) -> str:
+    """Insert b64-encoded string `chunk` into temporary document."""
+    collection = uploads_collection()
+
+    document = collection.get(key)
+    if document is None:
+        raise NonExistantUploadDocument()
+
+    document[sequence] = chunk
```

Comment:
Do we need to check whether there's already a chunk under this `sequence` ID?

Reply:
Yes, since having two documents with the same `sequence` ID would mean a corrupted file.